project-koku/koku
# [COST-3348] Update HCS finalization flow

## Jira Ticket

[COST-3348](https://issues.redhat.com/browse/COST-3348)

## Description

This change will require the finalization masu endpoint to specify both a month and a year if either are provided for date uniqueness. 
There is another change to check the `data_updated_timestamp` for a provider when we are getting providers to run HCS finalization on. Previously we would attempt to finalize any provider listed as "active". 

## Testing

1. Checkout Branch
2. Restart Koku
3. Load in some data with `make create-test-customer` and `make load-test-customer-data`
4. Try to run finalization on a specific month and a specific year separately through the masu endpoint and see a 400 response that indicates both are required.
5. http://localhost:5042/api/cost-management/v1/hcs_report_finalization/?year=2022
6. http://localhost:5042/api/cost-management/v1/hcs_report_finalization/?month=12
7. Try it again with both year and month and verify in the logs that finalization starts running, it will use the providers from the `get_accepted_providers` function if you don't specify a schema or provider_uuid.
8. http://localhost:5042/api/cost-management/v1/hcs_report_finalization/?year=2022&month=12

## Notes

...
